src/Groups/matrices/iso_nf_fq.jl

```diff
@@ -0,0 +1,172 @@
+# Detinko, Flannery, O'Brien "Recognizing finite matrix groups over infinite
+# fields", Section 4.2
+function isomorphic_group_over_finite_field(matrices::Vector{T}) where T <: Union{ fmpq_mat, AbstractAlgebra.MatElem{nf_elem} }
+   @assert !isempty(matrices)
+
+   # One should probably check whether all matrices are n by n (and invertible
+   # and such ...)
+
+   K = base_ring(matrices[1])
+   n = nrows(matrices[1])
+
+   Fq, matrices_Fq = good_reduction(matrices, 2)
+
+   G = MatrixGroup(n, Fq, matrices_Fq)
+   N = GAP.Globals.Order(G.X)
+   if maximal_order_of_finite_linear_group_qq(degree(K)*n) < N
+      error("Group is not finite")
+   end
+
+   G_to_fin_pres = GAP.Globals.IsomorphismFpGroupByGenerators(G.X, GAP.julia_to_gap([ g.X for g in gens(G) ]))
+   F = GAP.Globals.Range(G_to_fin_pres)
+   rels = GAP.Globals.RelatorsOfFpGroup(F)
+
+   inv_matrices = Vector{T}(undef, length(matrices))
+   inv_computed = falses(length(matrices))
+   for i = 1:length(rels)
```

Comment:
I think in this loop you are mapping the matrices through the relations. It should be possible to use `GAP.Globals.MappedWord` for that -- or just pass those matrices through the homomorphism

Reply:
I'm not sure what you mean. I have to evaluate the relations of the presentation at the matrices over the char 0 field. How could I do this with GAP? The matrices are in Nemo/AbstractAlgebra/Hecke.

Reply:
You can pass any Julia object into GAP functions; and all `MappedWord` cares about should be that the "generators" it gets can be multiplied and inverted. So it *should* work, if it doesn't, let me know and we can investigate why (but of course on the long run, there should be nice Julia wrappers for these anyway).

Reply:
I tried replacing it by
``` 
M = GAP.Globals.MappedWord(rels[i], GAP.Globals.GeneratorsOfGroup(F), GAP.julia_to_gap(matrices))
```
but that gives a "no method found!" error in GAP.
Also, we are potentially evaluating a lot of words here, so I would think it makes sense to cache the inverses of the matrices. But that wouldn't work with `MappedWord`.(?)

Reply:
You are right (although for now, I wouldn't worry about this kinda of performance issue too much: unless the matrices are gigantic, we won't notice compared to other steps of the algorithms). For now, I'd prefer clarity and conciseness. That said, if it works, it works; so  don't worry about this, we can revisit this in the future.

Reply:
Oh, and perhaps try again but with` FreeGeneratorsOfFpGroup` instead of `GeneratorsOfGroup`

Reply:
(as the relators are not elements of the fp group, but rather of the free group of which it is a quotient)

Reply:
That works!

Reply:
Okay, no, it doesn't.
```
julia> K, a = CyclotomicField(3, "a")
(Cyclotomic field of order 3, a)

julia> M1 = matrix(K, [ 0 0 1; 1 0 0; 0 1 0 ])
[0   0   1]
[1   0   0]
[0   1   0]

julia> M2 = matrix(K, [ 1 0 0; 0 a 0; 0 0 a^2 ])
[1   0        0]
[0   a        0]
[0   0   -a - 1]

julia> Oscar.isomorphic_group_over_finite_field([ M1, M2 ])
ERROR: Error thrown by GAP: Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `InverseMutable' on 1 arguments at /home/schmitt/.julia/artifacts/58582b7ec32af134aa4365459b9f929fa66b2087/share/gap/lib/methsel2.g:249 called from
InverseOp( elm ) at /home/schmitt/.julia/artifacts/58582b7ec32af134aa4365459b9f929fa66b2087/share/gap/lib/arith.gi:379 called from
Inverse( m ) at /home/schmitt/.julia/artifacts/58582b7ec32af134aa4365459b9f929fa66b2087/share/gap/lib/wordlett.gi:651 called from
<function "MappedWord for a letter assoc. word, a homogeneous list, and a list">( <arguments> )
 called from read-eval loop at *stdin*:1
```
It's possible that I never had the case that a relation contains an inverse before.

Reply:
With GAP.jl 0.6, I *think* this should now work...